src/Dfc.CourseDirectory.Core/DataStore/Sql/BulkCopyHelper.cs

```diff
@@ -0,0 +1,50 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Data;
+using System.Data.SqlClient;
+using System.Linq;
+using System.Threading.Tasks;
+using FastMember;
+
+namespace Dfc.CourseDirectory.Core.DataStore.Sql
+{
+    internal static class BulkCopyHelper
+    {
+        public static async Task WriteRecords<T>(IEnumerable<T> records, string tableName, SqlTransaction transaction)
```

Comment:
You could make this method easier to parse by extracting out a few well named methods, that way the overall steps at play here would stand out.